### ISSUE TITLE
Type constraint in foreign import error is not the case if a type alias is used

### ISSUE THREAD

**Issue by andys8** [purescript/purescript#4365]
## Description

Type constraints can't be used in foreign import and will lead to `ErrConstraintInForeignImportSyntax`. If a type alias is used for the same type (with constraint) the code compiles instead, but shouldn't.

## To Reproduce

This compiles.

```purescript
type T = forall a b. Union a b Props => ReactComponent (Record a)
foreign import comp :: T
```

This doesn't (inlining the type alias)

```purescript
foreign import comp :: forall a b. Union a b Props => ReactComponent (Record a)
```
> Constraints are not allowed in foreign imports. Omit the constraint instead

## Expected behavior

A type alias shouldn't make a difference and result in the same error.

## Additional context

The error is currently thrown in the parser. With a type alias this is a bit trickier. I still reported since and assume the behavior should be the same.

## PureScript version

0.15.3

**Comment by JordanMartinez** [comment]
I don't believe this is a bug. FFI aren't supposed to have type class constraints. The below code...
```purescript
foreign import showStuff :: forall a. Show a => a -> String
```
... becomes this in JavaScript:
```js
export const showStuff = (showDict) => (a) => ...
```
You'll need to remove the constraint by doing something like this
```purs
foreign import showStuffImpl :: forall a. (a -> String) -> a -> String

showStuff :: forall a. Show a => a -> String
showStuff a = showStuffImpl show a
```

**Comment by natefaubion** [comment]
@JordanMartinez the bug is that it can be circumvented with a type alias. We should check for this after synonyms are expanded.

**Comment by JordanMartinez** [comment]
Ah! My mistake. Thanks for clarifying that.
